fuel-tx/src/transaction/validity/error.rs

```diff
@@ -6,14 +6,10 @@ use fuel_types::{
     ContractId,
     MessageId,
 };
-#[cfg(feature = "std")]
-use std::{
-    error,
-    io,
-};
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
```

Comment:
Do we really need to support this kind of conversion? Because on the `fuel-core` side we also should be able to compile executor in `no_std` mode

Reply:
No, just avoiding breaking changes whenever possible. Do you think it whould be removed?